python-checks/src/test/resources/checks/argumentType_overloaded_functions.py

```diff
@@ -0,0 +1,42 @@
+from mod import predicate
+
+class A:
+  def make_a(self): ...
+
+class B:
+  def make_b(self): ...
+
+if predicate():
+  def overloaded_foo(param: A) : ...
+#     ^^^^^^^^^^^^^^>
+else:
+  def overloaded_foo(param: B): ...
+#     ^^^^^^^^^^^^^^>
+
+overloaded_foo(42)  # Noncompliant {{Change this argument; Function "overloaded_foo" expects a different type}}
+#              ^^
+
+
+if predicate():
+  def overloaded_bar(param1: A, param2: A): ...
+else:
+  def overloaded_bar(param1: B, param2: B): ...
+
+# we report only on the first noncompliant argument
+overloaded_bar(42, 42)  # Noncompliant
+#              ^^
+
+if predicate():
+  def overloaded_baz(param1: A): ...
+else:
+  def overloaded_baz(param1: B): ...
+
+overloaded_baz(A())  # OK
+
+
+if predicate():
+  def overloaded_fn(param1: A, param2: A): ...
+else:
+  def overloaded_fn(param1: B, param2: B): ...
+
+overloaded_fn(A(), B())  # Noncompliant
```

Comment:
Very good point, thanks! I finally decided to raise two separate issues (one for each argument), mainly because it was not easy to have a deterministic primary location, as it depends on the order to alternatives in the AmbiguousSymbol.
Let me know if that's fine for you.
Also, actually with `fn` I meant `function` instead of `false negative` 😅  I can change it if it's confusing.

Reply:
To be honest I think it's a bit weird to raise multiple issues, as it's really the combination of all argument types that create the issue.
Perhaps we could keep only the first noncompliant argument to appear syntactically for the primary location (by comparing token lines/columns) ? We might also highlight the other arguments for secondary locations.